engine/src/main/java/org/destinationsol/game/ship/hulls/Hull.java

```diff
@@ -158,22 +159,42 @@ public void onRemove(SolGame game) {
         if (myEngine != null) {
             myEngine.onRemove(game, myPos);
         }
+        myParticleEmitters.forEach(pe -> pe.onRemove(game, myPos));
 
     }
 
-    public void setEngine(SolGame game, SolShip ship, Engine ei) {
+    public void setEngine(SolGame game, SolShip ship, Engine engine) {
         List<Drawable> drawables = ship.getDrawables();
         if (myEngine != null) {
-            List<Drawable> dras1 = myEngine.getDrawables();
-            drawables.removeAll(dras1);
-            game.getDrawableManager().removeAll(dras1);
+            List<Drawable> engineDrawables = myEngine.getDrawables();
+            drawables.removeAll(engineDrawables);
+            game.getDrawableManager().removeAll(engineDrawables);
             myEngine = null;
         }
-        if (ei != null) {
-            myEngine = new ShipEngine(game, ei, config.getE1Pos(), config.getE2Pos(), ship);
-            List<Drawable> dras1 = myEngine.getDrawables();
-            drawables.addAll(dras1);
-            game.getDrawableManager().addAll(dras1);
+        if (engine != null) {
+            myEngine = new ShipEngine(game, engine, config.getE1Pos(), config.getE2Pos(), ship);
+            List<Drawable> engineDrawables = myEngine.getDrawables();
+            drawables.addAll(engineDrawables);
+            game.getDrawableManager().addAll(engineDrawables);
+        }
+    }
+
+    public void setParticleEmitters(SolGame game, SolShip ship, Engine engine) {
+        List<Drawable> drawables = ship.getDrawables();
+        if (!myParticleEmitters.isEmpty()) {
+            List<Drawable> peDrawables = new ArrayList<>();
```

Comment:
How about renaming this to `particleEmitterDrawables`? I know it's verbose, but I kinda like that over `peDrawables`..
I'm not sure about this one though.